sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -711,6 +712,23 @@ object DataSource extends Logging {
     }
   }
 
+  /**
+   * Return a [[TableProvider]] instance for the given provider. It returns None if the provider
+   * is not Data Source V2, or it's configured to fallback to Data Source V1 code path.
+   */
+  def lookupDataSourceV2(provider: String, conf: SQLConf): Option[TableProvider] = {
+    val useV1Sources = conf.getConf(SQLConf.USE_V1_SOURCE_LIST).toLowerCase(Locale.ROOT)
+      .split(",").map(_.trim)
+    val cls = lookupDataSource(provider, conf)
+    cls.newInstance() match {
+      case d: DataSourceRegister if useV1Sources.contains(d.shortName()) => None
```

Comment:
I know this is following the previous logic.
But what if the configuration `USE_V1_SOURCE_LIST` contains the canonical name of the `DataSourceRegister` class.  Maybe we can check this as well.

Reply:
It's checked in the next line.